multinode_integration_tests/tests/verify_bill_payment_utils/utils.rs

```diff
@@ -0,0 +1,1003 @@
+        vec![
+            "0x0bd8bc4b8aba5d8abf13ea78a6668ad0e9985ad6",
+            "0xb329c8b029a2d3d217e71bc4d188e8e1a4a8b924",
+            "0xb45a33ef3e3097f34c826369b74141ed268cdb5a",
+        ]
+        .iter()
+        .zip(self.serving_nodes.iter())
+        .for_each(|(expected_wallet_addr, serving_node)| {
+            let serving_node_actual = serving_node.earning_wallet.to_string();
+            assert_eq!(
+                &serving_node_actual,
+                expected_wallet_addr,
+                "{:?} wallet {} mismatched with expected {}",
+                serving_node.serving_node_profile.serving_node_by_name,
+                serving_node_actual,
+                expected_wallet_addr
+            );
+        })
+    }
+
+    fn assert_payments_via_direct_blockchain_scanning(&self, assertions_values: &AssertionsValues) {
+        let blockchain_interface = self
+            .global_values
+            .blockchain_interfaces
+            .standard_blockchain_interface
+            .as_ref();
+        assert_balances(
+            &self.consuming_node.consuming_wallet,
+            blockchain_interface,
+            assertions_values.final_consuming_node_transaction_fee_balance_minor,
+            assertions_values.final_consuming_node_service_fee_balance_minor,
+        );
+        assertions_values
+            .final_service_fee_balances_by_serving_nodes
+            .balances
+            .into_iter()
+            .zip(self.serving_nodes.iter())
+            .for_each(|(expected_remaining_owed_value, serving_node)| {
+                assert_balances(
+                    &serving_node.earning_wallet,
+                    blockchain_interface,
+                    0,
+                    expected_remaining_owed_value,
+                );
+            })
+    }
+
+    fn assert_serving_nodes_addressed_received_payments(
+        &self,
+        assertions_values: &AssertionsValues,
+    ) {
+        let actually_received_payments = assertions_values
+            .final_service_fee_balances_by_serving_nodes
+            .balances;
+        let consuming_node_wallet = &self.consuming_node.consuming_wallet;
+        self.serving_nodes
+            .iter()
+            .zip(actually_received_payments.into_iter())
+            .for_each(|(serving_node, received_payment)| {
+                let original_debt = serving_node.serving_node_profile.debt_specs().balance_minor;
+                let expected_final_balance = original_debt - received_payment;
+                Self::wait_for_exact_balance_in_receivables(
+                    &serving_node.receivable_dao,
+                    expected_final_balance,
+                    consuming_node_wallet,
+                )
+            })
+    }
+
+    fn wait_for_exact_balance_in_receivables(
+        node_receivable_dao: &ReceivableDaoReal,
+        expected_value: u128,
+        consuming_node_wallet: &Wallet,
+    ) {
+        test_utils::wait_for(Some(1000), Some(15000), || {
```

Comment:
Just by the looks of it, these values appears higher than what we might expect. Maybe you can look Into how this function works and maybe lower them to a more appropriate values.